### ISSUE TITLE
FALSE NEGATIVE: `TLW_TWO_LOCK_WAIT` misses a wait while two distinct monitors are held

### ISSUE THREAD

**Issue by rikki322** [spotbugs/spotbugs#4323]
**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs TLW_TWO_LOCK_WAIT`

**Minimal reproducer**

```java
class SpotbugsTlwTwoLockWaitDistinctMonitors {
  private final Object first = new Object();
  private final Object second = new Object();

  void waitWithTwoLocks() throws InterruptedException {
    synchronized (first) {
      synchronized (second) {
        second.wait();
      }
    }
  }
}
```

**Reproduction command**

```bash
javac -d classes spotbugs-tlw-two-lock-wait-fn-distinct-monitors.java
spotbugs -textui -effort:max -low -xml:withMessages -output spotbugs.xml classes
```

**Current behavior**

Two independent runs produce no `TLW_TWO_LOCK_WAIT` finding. SpotBugs may also report `UW_UNCOND_WAIT` and `WA_NOT_IN_LOOP`, but neither finding identifies the risk of retaining `first` while waiting on `second`.

**Expected behavior**

SpotBugs should report `TLW_TWO_LOCK_WAIT` at `second.wait()` because the call is reached while the two distinct monitors `first` and `second` are held. As documented by `TLW_TWO_LOCK_WAIT`, `wait()` releases only the monitor of the object being waited on (`second` here), while the other monitor (`first`) remains held.
